composer/cli/launcher.py

```diff
@@ -5,14 +5,14 @@
 import signal
 import subprocess
 import sys
-import tempfile
+import textwrap
 import time
 from argparse import ArgumentParser
-from typing import Any, List, Set
+from typing import Any, List, Optional, Set
 
 import torch.distributed
 
-CLEANUP_TIMEOUT = datetime.timedelta(seconds=5)
+CLEANUP_TIMEOUT = datetime.timedelta(seconds=30)
```

Comment:
Admittedly I remember finding that the processes never seemed to exit from `SIGTERM` within 5 seconds, but I assumed they were just hanging/non-responsive, not that they needed more time. Did this fix it?

Reply:
If wandb has artifacts to upload on a failed experiment, I want to give it slightly longer than 5 seconds.

Re: never exiting. We forgot to add a `.poll()` in there, so the `returncode` was never updated. So it never appeared to exit :rofl:

Reply:
Ah I see, that makes sense. XD